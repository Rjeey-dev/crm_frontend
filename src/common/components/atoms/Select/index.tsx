import React from 'react';

import {ReactNode, ReactNodeArray} from "react";

interface IProps {
    children: ReactNode | ReactNodeArray,
    classes?: string,
    name?: string
}

function Select(props: IProps) {
    return <select name={props.name} className={props.classes}>
        {props.children}
    </select>;
}

export default Select;