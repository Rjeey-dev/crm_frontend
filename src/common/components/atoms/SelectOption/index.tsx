import React from 'react';

export interface IOption {
    value: string,
    text: string,
    selected?: boolean
}

function SelectOption(props: IOption) {
    return <option selected={props.selected} value={props.value}>{props.text}</option>;
}

export default SelectOption;