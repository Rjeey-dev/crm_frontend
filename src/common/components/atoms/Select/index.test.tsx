import { shallow } from 'enzyme';
import React from 'react';

import SelectOption from '../SelectOption/index';
import Select from './index';

it("Renders <Select/> with correct properties", () => {
    const output = shallow(
        <Select classes='class'>
            <SelectOption value='value1' text='text1'/>
            <SelectOption value='value2' text='text2'/>
            <SelectOption value='value3' text='text3'/>
            <SelectOption value='value4' text='text4'/>
        </Select>
    ).html();

    expect(output).toMatchSnapshot();
});