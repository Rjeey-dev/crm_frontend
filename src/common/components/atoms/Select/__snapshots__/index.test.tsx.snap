// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Select/> with correct properties 1`] = `"<select class=\\"class\\"><option value=\\"value1\\">text1</option><option value=\\"value2\\">text2</option><option value=\\"value3\\">text3</option><option value=\\"value4\\">text4</option></select>"`;
